lib/puma/binder.rb

```diff
@@ -290,6 +290,19 @@ def add_tcp_listener(host, port, optimize_for_latency=true, backlog=1024)
 
       host = host[1..-2] if host and host[0..0] == '['
       tcp_server = TCPServer.new(host, port)
+
+      if defined? Localhost::Authority
+        raise "Puma compiled without SSL support" unless HAS_SSL
+        if ENV['RACK_ENV'] == "development" || ENV['RACK_ENV'] == "test"
+          authority = Localhost::Authority.fetch
```

Comment:
Try to do this in the top level process (I assume this is) because there are some race conditions which I need to fix - if multiple processes try to get the same certificate at the same time, they may clobber each other - in theory not a big problem (especially for development) but you can simply avoid it by ensuring it's only called once per session and at some point I'll make sure it's more robust.

Reply:
By the way in several years of using It, I've not even had a single bug report about it. So, it's kind of a non issue in practice.

Reply:
Okay, I fixed it :) Still do it in the top level process to avoid generating a different certificate per worker if possible :)